### PR TITLE
Add Table Story

### PR DESCRIPTION
**Pull request by guergana**
This PR creates a Story for the Table component added in [this commit](https://github.com/wmde/wikit/commit/350677c6d2704db7480cfdf2a619b847a44e61e8).

Bug: [T291077](https://phabricator.wikimedia.org/T291077)

### CODE REVIEW

File: vue-components/src/components/Table.vue [hunk 1]
```diff
@@ -66,30 +65,30 @@ export default Vue.extend( {
...
-		*   Column Header: Content Here
+		*   <td data-header="Column Header">Content Here</td>
 		*/
 		/* stylelint-disable selector-no-qualifying-type  */
 		th[data-header]::before,
 		td[data-header]::before {
 			content: attr(data-header);
 			display: block;
-			font-weight: bold;
+			font-weight: $font-weight-bold;
```
Comment (SaiSan-WMDE) [inline vue-components/src/components/Table.vue:80]:
```suggestion
			font-weight: $wikit-Table-cell-th-font-weight;
```

File: vue-components/src/components/Table.vue [hunk 2]
```diff
@@ -66,30 +65,30 @@ export default Vue.extend( {
...
 			flex-basis: 40%;
+			// Ensure headers stay exactly 40%
+			// even if values are wider than 60%
+			min-width: 40%;
 		}
 
 		th:not([data-header]) {
-			font-weight: bold;
+			font-weight: $font-weight-bold;
```
Comment (SaiSan-WMDE) [inline vue-components/src/components/Table.vue:88]:
```suggestion
			font-weight: $wikit-Table-cell-th-font-weight;
```

File: vue-components/stories/Table.stories.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                        </tr>
+                    </thead>
+                    <tbody>
+                        <tr v-for="cell in cells" :key="cell.col1">
+                            <!-- When using the table make sure to include the
+                                data-header for linearization and accesibility -->
+                            <td data-header="Vegetable">
+                                {{cell.col1}}
+                            </td>
+                            <td data-header="Column 2">{{cell.col2}}</td>
```
Comment (SaiSan-WMDE) [inline vue-components/stories/Table.stories.ts:73]:
```suggestion
                            <td data-header="Column 2">{{cells.col2}}</td>
```

Comment (SaiSan-WMDE) [reply vue-components/stories/Table.stories.ts:73]: I think this is what may be causing the headers to display the wrong content when the table is linearized?

File: vue-components/stories/Table.stories.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                            <td data-header="Column 3">{{cell.col3}}</td>
```
Comment (SaiSan-WMDE) [inline vue-components/stories/Table.stories.ts:74]:
```suggestion
                            <td data-header="Column 3">{{cells.col3}}</td>
```

File: vue-components/stories/Table.stories.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+                            <td data-header="Column 4">{{cell.col4}}</td>
```
Comment (SaiSan-WMDE) [inline vue-components/stories/Table.stories.ts:75]:
```suggestion
                            <td data-header="Column 4">{{cells.col4}}</td>
```

File: vue-components/stories/Table.stories.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+                            <th scope="col">Calories</th>
+                            <th scope="col">Amount</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        <tr v-for="cell in cells" :key="cell.col1">
+                            <!-- When using the table make sure to include the
+                                data-header for linearization and accesibility -->
+                            <td data-header="Vegetable">
+                                {{cell.col1}}
```
Comment (SaiSan-WMDE) [inline vue-components/stories/Table.stories.ts:71]:
```suggestion
                                {{cells.col1}}
```

File: vue-components/stories/Table.stories.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+                            <th scope="col">Cooked</th>
+                            <th scope="col">Calories</th>
+                            <th scope="col">Amount</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        <tr v-for="cell in cells" :key="cell.col1">
+                            <!-- When using the table make sure to include the
+                                data-header for linearization and accesibility -->
+                            <td data-header="Vegetable">
```
Comment (SaiSan-WMDE) [inline vue-components/stories/Table.stories.ts:70]: Should we set "Column 1" as the value of "data-header" here, to be consistent? Not sure how important this is

File: vue-components/stories/Table.stories.ts (hunk 3 above)
Comment (itamargiv) [reply vue-components/stories/Table.stories.ts:73]: The value of data header should be displayed, this is how the table works...

Comment (SaiSan-WMDE) [reply vue-components/stories/Table.stories.ts:73]: ok. then this was a misunderstanding

Comment (itamargiv) [reply vue-components/stories/Table.stories.ts:73]: But of course we need to document this more properly, and use the correct column headers

File: vue-components/src/components/Table.vue [hunk 8]
```diff
@@ -14,9 +14,33 @@
...
 
+/**
+ * Tables display categorical information organized across rows and columns in
+ * order to facilitate the comparative analysis of data.
+ *
+ * The WiKit table component provides a wrapper around the common HTML table
+ * elements such as `<thead>`, `<tbody>`, `<tr>`, `<th>` and `<td>`, to apply
+ * design system styles to tabular data.
+ *
+ * Adding a `data-header` attribute to the cells allows us to maintin the column
```
Comment (itamargiv) [inline vue-components/src/components/Table.vue:25]:
```suggestion
 * Adding a `data-header` attribute to the cells allows us to maintain the column
```

File: vue-components/src/components/Table.vue [hunk 9]
```diff
@@ -14,9 +14,33 @@
...
+ * headers and display them in the table's linearzed form to provide additional
```
Comment (itamargiv) [inline vue-components/src/components/Table.vue:26]:
```suggestion
 * headers and display them in the table's linearized form to provide additional
```

Comment (Silvan-WMDE) [inline vue-components/src/components/Table.vue:26]:
```suggestion
 * headers and display them in the table's linearized form to provide additional
```

File: vue-components/src/components/Table.vue [hunk 10]
```diff
@@ -66,30 +89,30 @@ export default Vue.extend( {
...
 		th[data-header]::before,
 		td[data-header]::before {
 			content: attr(data-header);
 			display: block;
-			font-weight: bold;
+			font-weight: $wikit-Table-cell-th-font-weight;
 			flex-basis: 40%;
+			// Ensure headers stay exactly 40%
+			// even if values are wider than 60%
+			min-width: 40%;
```
Comment (Silvan-WMDE) [inline vue-components/src/components/Table.vue:108]: :+1:

**Review by Silvan-WMDE — COMMENTED** [review]
Looks good to me and works great on [chromatic](https://5efdb3b5f65950002286285d-uajkkxqjsn.chromatic.com/?path=/story/table--basic)

**Review by Silvan-WMDE — APPROVED** [review]

File: vue-components/src/components/Table.vue [hunk 11]
```diff
@@ -66,30 +89,30 @@ export default Vue.extend( {
...
 		}
 
 		th:not([data-header]) {
-			font-weight: bold;
+			font-weight: $wikit-Table-cell-th-font-weight;
```
Comment (SaiSan-WMDE) [inline vue-components/src/components/Table.vue:112]:
```suggestion
			font-weight: $wikit-Table-cell-heading-font-weight;
```

File: vue-components/src/components/Table.vue [hunk 12]
```diff
@@ -66,30 +89,30 @@ export default Vue.extend( {
...
-		*   Column Header: Content Here
+		*   <td data-header="Column Header">Content Here</td>
 		*/
 		/* stylelint-disable selector-no-qualifying-type  */
 		th[data-header]::before,
 		td[data-header]::before {
 			content: attr(data-header);
 			display: block;
-			font-weight: bold;
+			font-weight: $wikit-Table-cell-th-font-weight;
```
Comment (SaiSan-WMDE) [inline vue-components/src/components/Table.vue:104]:
```suggestion
			font-weight: $wikit-Table-cell-heading-font-weight;
```

File: vue-components/src/components/Table.vue [hunk 13]
```diff
@@ -163,7 +186,7 @@ export default Vue.extend( {
 			/**
 			* Typography
 			*/
-			font-weight: $font-weight-bold;
+			font-weight: $wikit-Table-cell-th-font-weight;
```
Comment (SaiSan-WMDE) [inline vue-components/src/components/Table.vue:189]:
```suggestion
			font-weight: $wikit-Table-cell-heading-font-weight;
```

**Comment by itamargiv** [comment]
What an all encompassing collaboration! Well done everyone :clap: :clap:

**Review by SaiSan-WMDE — APPROVED** [review]
💯

**Pull request merged**